### PR TITLE
Letting the user override the code generation a bit.

### PR DESCRIPTION
**Pull request by nicolasnoble**
Example of use:

protoc --grpc_out=use_system_headers=false,grpc_search_path=a/b/c/d:path/to/output/...

### CODE REVIEW

File: src/compiler/cpp_generator.cc [hunk 1]
```diff
@@ -702,13 +717,23 @@ grpc::string GetSourceIncludes(const grpc::protobuf::FileDescriptor *file,
...
     std::map<grpc::string, grpc::string> vars;
 
-    printer.Print(vars, "#include <grpc++/channel.h>\n");
-    printer.Print(vars, "#include <grpc++/impl/client_unary_call.h>\n");
-    printer.Print(vars, "#include <grpc++/impl/rpc_service_method.h>\n");
-    printer.Print(vars, "#include <grpc++/impl/service_type.h>\n");
-    printer.Print(vars, "#include <grpc++/support/async_unary_call.h>\n");
-    printer.Print(vars, "#include <grpc++/support/async_stream.h>\n");
-    printer.Print(vars, "#include <grpc++/support/sync_stream.h>\n");
+    vars["l"] = params.use_system_headers ? '<' : '"';
```
Comment (ctiller) [inline src/compiler/cpp_generator.cc]: Could you factor this out into a:

void PrintHeaders(grpc::protobuf::io::Printer *printer, const std::vector<grpc::string>& headers);

Comment (ctiller) [inline src/compiler/cpp_generator.cc]: (it's going to make future changes a touch easier)

**Comment by nicolasnoble** [comment]
Merged from master and taking comments into account.

**Comment by nicolasnoble** [comment]
There's a new flake I haven't seen before. https://grpc-testing.appspot.com/job/gRPC_pull_requests/7050/config=opt,language=c,platform=linux/testReport/junit/(root)/tests/bins_opt_lb_policies_test/

**Comment by nicolasnoble** [comment]
@yang-g ping

**Pull request merged**